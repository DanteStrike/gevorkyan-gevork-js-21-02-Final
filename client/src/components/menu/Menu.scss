@import "src/assets/variables";

.menu {
  display: flex;
  align-items: center;
  justify-content: space-between;
  width: 305px;
  height: 50px;
  padding: 0;
  margin: 0;
  color: var(--primary-font-color);
  background-color: var(--primary-background);
  border-color: var(--primary-background);

  &__link {
    display: inline;
    padding: 0;
  }

  &__item {
    font-family: var(--secondary-font-family);
    font-size: 18px;
  }
}

@media (max-width: $tabletWidth) {
  .menu {
    height: 45px;

    &__item {
      font-size: 16px;
    }
  }
}

.ant-menu-horizontal > .ant-menu-item a {
  color: var(--primary-font-color);
}
