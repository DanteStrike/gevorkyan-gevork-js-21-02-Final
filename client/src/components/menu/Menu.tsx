import React from 'react';
import {Menu as MenuANTD} from 'antd';
import {UserOutlined, PictureOutlined} from '@ant-design/icons';
import CustomLink from '../custom-link/CustomLink';
import './Menu.scss';
import {RoutePath} from '../../enums';
import useAppTranslation from '../../hooks/use-app-translation';

const iconStyle = {
  fontSize: `20px`,
};

function Menu() {
  const {t} = useAppTranslation(`menu`);

  return (
    <MenuANTD className="menu" mode="horizontal" selectedKeys={[]}>
      <MenuANTD.Item className="menu__item" key="users" icon={<UserOutlined style={iconStyle} />}>
        <CustomLink className="menu__link" to={RoutePath.USERS}>
          <span className="menu__text">{t(`users`)}</span>
        </CustomLink>
      </MenuANTD.Item>
      <MenuANTD.Item className="menu__item" key="posts" icon={<PictureOutlined style={iconStyle} />}>
        <CustomLink className="menu__link" to={RoutePath.POSTS}>
          <span className="menu__text">{t(`posts`)}</span>
        </CustomLink>
      </MenuANTD.Item>
    </MenuANTD>
  );
}

export default React.memo(Menu);
