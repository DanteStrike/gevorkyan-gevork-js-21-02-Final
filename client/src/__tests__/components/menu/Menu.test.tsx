import {shallow} from 'enzyme';
import React from 'react';
import Menu from '../../../components/menu/Menu';

describe(`Component Menu should render correctly`, () => {
  it(`snapshot default test`, () => {
    const wrapper = shallow(<Menu />);
    expect(wrapper).toMatchSnapshot();
  });
});
