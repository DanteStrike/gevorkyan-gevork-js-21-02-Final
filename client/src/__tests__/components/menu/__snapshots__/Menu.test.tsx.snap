// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Menu should render correctly snapshot default test 1`] = `
<Menu
  className="menu"
  mode="horizontal"
  selectedKeys={Array []}
>
  <MenuItem
    className="menu__item"
    icon={
      <ForwardRef(UserOutlined)
        style={
          Object {
            "fontSize": "20px",
          }
        }
      />
    }
    key="users"
  >
    <Memo(CustomLink)
      className="menu__link"
      to="/users"
    >
      <span
        className="menu__text"
      >
        users
      </span>
    </Memo(CustomLink)>
  </MenuItem>
  <MenuItem
    className="menu__item"
    icon={
      <ForwardRef(PictureOutlined)
        style={
          Object {
            "fontSize": "20px",
          }
        }
      />
    }
    key="posts"
  >
    <Memo(CustomLink)
      className="menu__link"
      to="/posts"
    >
      <span
        className="menu__text"
      >
        posts
      </span>
    </Memo(CustomLink)>
  </MenuItem>
</Menu>
`;
